.github/workflows/test.yml

```
@@ -0,0 +1,66 @@
+on:
+  pull_request:
+    branches:
+      - 'polkadot-v0.9.42'
+      - 'master'
+
+name: continuous-integration-test
+jobs:
+  ci:
+    runs-on: ubuntu-20.04
+    env:
+      RUST_BACKTRACE: full
+      # Make sure CI fails on all warnings, including Clippy lints
+      RUSTFLAGS: "-Dwarnings"
+
+    steps:
+      - name: Freeing up more disk space
+        run: |
+          free -h
+          sudo rm -rf /usr/local/lib/android # will release about 10 GB if you don't need Android
+          sudo rm -rf /usr/share/dotnet # will release about 20GB if you don't need .NET
+          sudo rm -rf /opt/ghc
+          sudo rm -rf /usr/local/share/boost
+          sudo rm -rf /opt/hostedtoolcache/CodeQL
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+          sudo apt-get remove -y 'php.*' --fix-missing
+          sudo apt-get remove -y '^mongodb-.*' --fix-missing
+          sudo apt-get remove -y '^mysql-.*' --fix-missing
+          sudo apt-get clean          
+          df -h
+      - uses: actions/checkout@v2
+
+      - name: Install package
+        run: |
+          sudo docker image prune --all --force
+          echo 'APT::Get::Always-Include-Phased-Updates "false";' | sudo tee /etc/apt/apt.conf.d/99-phased-updates
+          sudo apt-get update && sudo apt-get upgrade -y
+          sudo apt-get install -y protobuf-compiler libprotobuf-dev
+
+      - name: Setup Rust toolchain
+        # Call `rustup show` as a hack so that the toolchain defined in rust-toolchain.toml is installed
+        run: rustup show
+
+      - name: Use Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+          key: "ubuntu-latest-cargo-${{ hashFiles('**/Cargo.lock') }}"
```

Nit: In line 10, we didn't define `ubuntu-latest` but `ubuntu-20.04` so this key should be
```suggestion
          key: "ubuntu-20.04-cargo-${{ hashFiles('**/Cargo.lock') }}"
```